pkg/booleanpolicy/default_policies_test.go

```diff
@@ -2272,6 +2276,82 @@ func (suite *DefaultPoliciesTestSuite) TestImageVerified() {
 	}
 }
 
+func (suite *DefaultPoliciesTestSuite) TestImageVerified_WithDeployment() {
+	const (
+		verifier1 = "io.stackrox.signatureintegration.00000000-0000-0000-0000-000000000002"
+		verifier2 = "io.stackrox.signatureintegration.00000000-0000-0000-0000-000000000003"
+		verifier3 = "io.stackrox.signatureintegration.00000000-0000-0000-0000-000000000004"
+	)
+
+	images := []*storage.Image{
+		imageWithSignatureVerificationResults("image_verified_by_1", []*storage.ImageSignatureVerificationResult{
+			{
+				VerifierId:              verifier1,
+				Status:                  storage.ImageSignatureVerificationResult_VERIFIED,
+				VerifiedImageReferences: []string{"image_verified_by_1"},
+			},
+		}),
+		imageWithSignatureVerificationResults("image_with_alternative_verified_reference", []*storage.ImageSignatureVerificationResult{
+			{
+				VerifierId:              verifier2,
+				Status:                  storage.ImageSignatureVerificationResult_VERIFIED,
+				VerifiedImageReferences: []string{"image_verified_by_2"},
+			},
+		}),
+		imageWithSignatureVerificationResults("image_verified_by_2", []*storage.ImageSignatureVerificationResult{
+			{
+				VerifierId:              verifier3,
+				Status:                  storage.ImageSignatureVerificationResult_VERIFIED,
+				VerifiedImageReferences: []string{"image_with_alternative_verified_reference", "image_verified_by_2"},
+			},
+		}),
+	}
+
+	cases := map[string]struct {
+		deployment       *storage.Deployment
+		image            *storage.Image
+		matchingVerifier string
+		expectViolation  bool
+	}{
+		"deployment with matching verified image reference shouldn't lead in alert message": {
+			deployment:       deploymentWithImage("deployment_with_image_verified_by_1", images[0]),
+			image:            images[0],
+			matchingVerifier: verifier1,
+		},
+		"deployment with verified result but no matching verified image reference should lead to alert message": {
+			deployment:       deploymentWithImage("deployment_with_image_alternative_verified_reference", images[1]),
+			image:            images[1],
+			matchingVerifier: verifier2,
+			expectViolation:  true,
+		},
+		"deployment with verified result and multiple matching verified image references shouldn't lead to alert message": {
+			deployment:       deploymentWithImage("deployment_with_image_verified_by_2", images[2]),
+			image:            images[2],
+			matchingVerifier: verifier3,
+		},
+	}
+
+	for name, c := range cases {
+		suite.Run(name, func() {
+			deploymentMatcher, err := BuildDeploymentMatcher(policyWithSingleFieldAndValues(fieldnames.ImageSignatureVerifiedBy,
+				[]string{c.matchingVerifier}, false, storage.BooleanOperator_OR))
+			suite.Require().NoError(err)
+
+			violations, err := deploymentMatcher.MatchDeployment(nil, EnhancedDeployment{
+				Deployment: c.deployment,
+				Images:     []*storage.Image{c.image},
+			})
+			suite.Require().NoError(err)
+
+			if c.expectViolation {
+				suite.NotEmpty(violations.AlertViolations)
```

Comment:
I do not remember the violation message format for unverified image signature alerts. Do we print the image name in the message? If so, is the correct image name in the violations in case of multiple names? Could you add some tests to cover it?

Reply:
Right now, we only print the container name in addition, and no actual image name in the case of scanning the image itself. So, no need to change anything there for now.

Overall, we've discussed improving this information as well (printing additional info, being more clear etc.), but haven't gotten the priority and customer feedback to actually change this.